gobblin-runtime/src/main/java/org/apache/gobblin/runtime/SafeDatasetCommit.java

```diff
@@ -169,6 +177,26 @@ public Void call()
     return null;
   }
 
+  public void submitLineageEvent(JobState.DatasetState datasetState) {
+    Collection<TaskState> states = datasetState.getTaskStates();
+    if (states.size() == 0) {
+      return;
+    }
+
+    TaskState oneWorkUnitState = states.iterator().next();
+    if (oneWorkUnitState.contains(ConfigurationKeys.JOB_DATA_PUBLISHER_TYPE) && oneWorkUnitState.getProp(ConfigurationKeys.JOB_DATA_PUBLISHER_TYPE).equals(
+        NoopPublisher.class.getName())) {
+      // if no publisher configured, each task should be responsible for sending lineage event.
+      return;
```

Comment:
Isn't it easier to do it here either way? What's the point of making the task do the work?